packages/kit/src/runtime/server/page/server_routing.js

```diff
@@ -0,0 +1,81 @@
+import { s } from '../../../utils/misc.js';
+import { has_server_load } from '../../../utils/routing.js';
+import { load_page_and_error_nodes } from './load_page_nodes.js';
+import { text } from '../../../exports/index.js';
+
+/**
+ * @param {URL} url
+ * @param {import('types').SSROptions} options
+ * @returns {boolean}
+ */
+export function is_route_resolution_request(url, options) {
+	return (
+		url.pathname === `/${options.app_dir}/routes.js` ||
+		url.pathname.startsWith(`/${options.app_dir}/routes/`)
+	);
+}
+
+/**
+ * @param {URL} url
+ * @param {import('types').SSROptions} options
+ * @returns {string}
+ */
+export function route_resolution_to_regular_route(url, options) {
+	return url.pathname.slice(`/${options.app_dir}/routes`.length, -3) || '/';
+}
+
+/**
+ * @param {URL} url
+ * @param {import('types').SSROptions} options
+ * @returns {string}
+ */
+export function regular_route_to_route_resolution(url, options) {
+	return `/${options.app_dir}/routes` + (url.pathname === '/' ? '.js' : url.pathname + '.js');
+}
+
+/**
+ * @param {string | null} id
+ * @param {import('types').PageNodeIndexes} page
+ * @param {import('@sveltejs/kit').SSRManifest} manifest
+ * @returns {Promise<string>}
+ */
+export async function create_stringified_csr_server_route(id, page, manifest) {
+	const { errors, layouts, leaf } = await load_page_and_error_nodes(page, manifest);
+
+	const nodes = `{ ${[...errors, ...layouts, leaf]
+		.filter((n) => !!n)
+		.map((n) => `'${n.index}': () => import('/${n.imports[0]}')`)
```

Comment:
can we make this relative, i.e. `import('../../immutable/nodes/1.abcdef.js')`? Otherwise we'll break apps that depend on relative paths